### PR TITLE
Self-service site reset: make the UI match the mockup

### PR DESCRIPTION
**Pull request by mpkelly**
Closes https://github.com/Automattic/dotcom-forge/issues/4779
Closes https://github.com/Automattic/dotcom-forge/issues/4781

**Atomic**
image

**Simple**

Screenshot 2566-12-07 at 17 40 33

## Proposed Changes

1. Show a list of content that will be removed
2. Only allow resetting if the domain test value matches the actual domain
3. Update the notice text to reflect the async nature of the Atomic reset
4. Refactors code so hooks now live in sub-package
5. Add a hint to the bottom of the card about Activity Log and Export https://github.com/Automattic/dotcom-forge/issues/4781
6. Make card space match mockup https://github.com/Automattic/dotcom-forge/issues/4781
7. Change text at to of card so it mentions theme changes. 

**Note**: there doesn't appear to be an easy way to detect user-installed plugins, so the plugin count includes pre-installed plugins. I think this might cause the user to panic. What do you think about just saying, "Plugins you installed"? Or maybe we can get the info, e.g., check the Activity Log instead?

## Testing Instructions

* Apply D130278-code and sandbox API
* Check out this branch
* Try with Simple and Atomic sites by visiting `/settings/start-over/:site`

This new UI needs some units tests but they can be added in https://github.com/Automattic/dotcom-forge/issues/4793

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by vykes-mac — APPROVED** [review]
This is g2g LGTM 🚀

File: client/my-sites/site-settings/start-over.jsx [hunk 1]
```diff
@@ -208,16 +278,20 @@ function SiteResetCard( { translate, selectedSiteSlug, siteDomain, isAtomic } )
 							id="confirmResetInput"
 							disabled={ isLoading }
 							style={ { flex: 0.5 } }
+							onChange={ ( event ) =>
+								setDomainConfirmed( event.currentTarget.value.trim() === siteDomain )
```
Comment (mpkelly) [inline client/my-sites/site-settings/start-over.jsx:282]: Nice solution, @vykes-mac.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10069626

Thank you @mpkelly for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10069626).

**Pull request merged**